lib/matplotlib/tests/test_patches.py

```diff
@@ -999,3 +999,79 @@ def test_set_and_get_hatch_linewidth(fig_test, fig_ref):
 
     assert ax_ref.patches[0].get_hatch_linewidth() == lw
     assert ax_test.patches[0].get_hatch_linewidth() == lw
+
+
+@check_figures_equal(extensions=['png', 'pdf', 'svg', 'eps'], tol=0.021)
```

Comment:
Why is there a non-zero tolerance?

Reply:
The slight RMS error comes when using the workaround way to plot patches with hatchcolor different from edgecolor. This error isn't visible in the diff image, so I am not sure what exactly the problem is.

However, I noticed that setting the linewidth as 2 or higher, instead of 1 no longer causes any RMS error.

So I think there are three ways to go about this:

1. Let the tolerance be non-zero
2. Use linewidth 2, and zero tolerance
3. Make this a image comparision test (as suggested [here](https://github.com/matplotlib/matplotlib/pull/28104#discussion_r1891236978))